pkg/registry/pod/rest.go

```diff
@@ -55,6 +55,9 @@ func (podStrategy) PrepareForCreate(obj runtime.Object) {
 	pod.Status = api.PodStatus{
 		Phase: api.PodPending,
 	}
+	for i := range pod.Spec.ImagePullSecrets {
+		pod.Spec.ImagePullSecrets[i] = api.ObjectReference{Name: pod.Spec.ImagePullSecrets[i].Name}
```

Comment:
> On May 18, 2015, at 6:15 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/registry/pod/rest.go:
> 
> > @@ -55,6 +55,9 @@ func (podStrategy) PrepareForCreate(obj runtime.Object) {
> >  	pod.Status = api.PodStatus{
> >  		Phase: api.PodPending,
> >  	}
> > +	for i := range pod.Spec.ImagePullSecrets {
> > +		pod.Spec.ImagePullSecrets[i] = api.ObjectReference{Name: pod.Spec.ImagePullSecrets[i].Name}
> Yes, documented here:
> #7018 (comment)
> 
> The only question is whether to fill in the other relevant ObjectReference fields or leave them empty. If you'd like to do the latter, how hard would it be to just create LocalObjectReference now? It seems simpler than verifying that the fields are empty and less confusing than simply throwing away their values (though without schema validation, json unmarshalling will do just that).
> 
> // Reference to another object within a namespace
> type LocalObjectReference struct {
> //TODO: Add apiVersion and kind
>     Name            string    `json:"name" description:"name of the referent"`
> }
That's fine with me - just wanted to give David the right tools to implement this.

> —
> Reply to this email directly or view it on GitHub.
>